library/src/main/java/com/novoda/downloadmanager/lib/OkHttpNotificationImageRetriever.java

```diff
@@ -28,7 +38,10 @@ public Bitmap retrieveImage(String imageUrl) {
             Response response = client.newCall(request).execute();
             InputStream inputStream = response.body().byteStream();
             try {
-                return BitmapFactory.decodeStream(inputStream);
+                this.imageUrl = imageUrl;
```

Comment:
So that we only set the new image URL on successful requests. If we set it at the start then the cached bitmap could be returned even after a failed request